ratelimits/gcra.go

```diff
@@ -77,16 +79,16 @@ func maybeRefund(clk clock.Clock, rl limit, tat time.Time, cost int64) *Decision
 	if nowUnix > tatUnix {
 		// The TAT is in the past, therefore the bucket is full.
 		return &Decision{
-			Allowed:   false,
-			Remaining: rl.Burst,
-			RetryIn:   time.Duration(0),
-			ResetIn:   time.Duration(0),
+			allowed:   false,
+			remaining: txn.limit.Burst,
+			retryIn:   time.Duration(0),
+			resetIn:   time.Duration(0),
```

Comment:
Are we missing `transaction : txn,` here?

Reply:
Ooof, good catch! I have also added a test to ensure we don't see a regression.